### PR TITLE
Jon/fix/new-user-fio-wallet-error-v2

### PR DESCRIPTION
**Pull request by Jon-edge**
https://user-images.githubusercontent.com/90650827/231905447-bfe182bc-a1ba-4981-91c8-634438a9cc2d.mp4

### CHANGELOG

- fixed: Error dropdown in some cases when creating a free FIO handle

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [x] Tested on iOS device
- [x] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

### CODE REVIEW

File: src/actions/LoginActions.tsx [hunk 1]
```diff
@@ -83,19 +83,20 @@ export function initializeAccount(navigation: NavigationBase, account: EdgeAccou
...
-          ]).then(async ([_customWallets, _fioCreate]) => {
-            return await updateWalletsRequest()(dispatch, getState)
-          })
-        } else {
-          await createCustomWallets(account, fiatCurrencyCode, selectedEdgetokenIds, dispatch)
-          await updateWalletsRequest()(dispatch, getState)
+            .catch(showError)
         }
+
+        await createWalletsPromise
```
Comment (paullinator) [inline src/actions/LoginActions.tsx:98]: Yikes. you'll be calling updateWalletsRequest before the fio wallets are created. I would await the Airship.show() instead and do away with .then()

File: src/components/modals/FioCreateHandleModal.tsx [hunk 2]
```diff
@@ -38,6 +43,22 @@ export const FioCreateHandleModal = (props: Props) => {
...
+    if (walletIds.length === 0) return
+
+    let sum = 0
+    for (const walletId of walletIds) {
+      sum += walletsForProgress[walletId]
+    }
+    let ratio = sum / walletIds.length
+    if (ratio > 0.99999) ratio = 1
+
+    console.debug(sum)
```
Comment (paullinator) [inline src/components/modals/FioCreateHandleModal.tsx:57]: remove

File: src/components/modals/FioCreateHandleModal.tsx [hunk 3]
```diff
@@ -38,6 +43,22 @@ export const FioCreateHandleModal = (props: Props) => {
     bridge.resolve(true)
   })
 
+  React.useEffect(() => {
```
Comment (paullinator) [inline src/components/modals/FioCreateHandleModal.tsx:46]: I don't think this % stuff is necessary and it's more regression risk too late in the release.

**Review by paullinator — CHANGES_REQUESTED** [review]

File: src/components/modals/FioCreateHandleModal.tsx [hunk 4]
```diff
@@ -25,6 +28,8 @@ export const FioCreateHandleModal = (props: Props) => {
   const styles = getStyles(theme)
   const isCreateWalletsInProgress = typeof createWalletsPromise === 'object' && createWalletsPromise instanceof Promise
   const [showPleaseWait, setShowPleaseWait] = React.useState(false)
+  const [completePercent, setCompletePercent] = React.useState('0%')
+  const walletsForProgress = useSelector(state => state.ui.wallets.walletLoadingProgress)
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/modals/FioCreateHandleModal.tsx:32]: I'm only waiting for createWalletPromise...

**Review by Jon-edge — COMMENTED** [review]
I reverted the last change for pct display, though I think it's still correct aside from the extra debug statement

File: src/components/modals/FioCreateHandleModal.tsx [hunk 5]
```diff
@@ -43,6 +50,7 @@ export const FioCreateHandleModal = (props: Props) => {
           {lstrings.fio_free_handle_message}
         </EdgeText>
       </View>
+      {showPleaseWait ? <EdgeText>{/* TODO: */}</EdgeText> : null}
```
Comment (paullinator) [inline src/components/modals/FioCreateHandleModal.tsx:53]: Need some text here.

**Review by paullinator — CHANGES_REQUESTED** [review]
Missing text to show when waiting

**Review by paullinator — APPROVED** [review]

**Pull request merged**